tests/v3/covid-19/api_historical.spec.js

```diff
@@ -109,15 +109,15 @@ describe('TESTING /v3/covid-19/historical', () => {
 		chai.request(app)
 			.get('/v3/covid-19/historical/usa/sdgdf,gsfd')
 			.end((err, res) => {
-				testBasicProperties(err, res, 200, 'array');
+				testBasicProperties(err, res, 404, 'array');
 				res.body[0].should.have.property('message');
 				done();
 			});
 	});
 
 	it('/v3/covid-19/historical/ correct province name list', (done) => {
 		chai.request(app)
-			.get('/v3/covid-19/historical/156/bejing,hubei')
+			.get('/v3/covid-19/historical/156/bejing,hebei')
```

Comment:
`bejing` is not a province in China (I changed it to `beijing` in my latest commit). In the endpoint, you are checking only the first element in the array. If it is not `null`, the response status code will be `200` but in this case because it is `null`, the `else` clause will be evaluated resulting in response status code of `404` though the second province has data. As a result the above test fails. I suggest you only return status code of 404 if all the provinces are null. Something like:
```js
if (countryData && (countryData.some(data => data !== null)) {
		res.send(countryData.length === 1 ? countryData[0] : countryData);
	} else 
```

Reply:
I see the issue, good catch. Do you want to fix since you already have the code here?

Basically just need this to be fixed and then we can merge your feature

Reply:
Well in the process of fixing the above code, I have also noticed that if  data for a single country is requested without a province or region, the data returned is not an array. From:
```js
countryData = scraper.historical.getHistoricalCountryDataV2(
			data,
			query,
			province,
			lastdays
		);
```
Essentially `countryData` is an object with the shape:
```js
{
  "country": "string",
  "province": [
    "string"
  ],
  "timeline": {
    "cases": {
      "date": 0
    },
    "deaths": {
      "date": 0
    },
    "recovered": {
      "date": 0
    }
  }
}
```
But then we are using the length of `countryData` in `res.send(countryData.length === 1 ? countryData[0] : countryData);` All along this wasn't noticed, I guess, is because `{}.length === 1` evaluates to `false` since `{}.length` is `undefined` hence returning `countryData`. So to fix the above I have decided to use `countries.map` in the `else` clause where `countryData` is assigned a value so that it will always be an array. You can find it in my latest commit.

Reply:
Good find, I think this is ready to merge yeah? 😄

Reply:
I believe so.